runtime/vm/jvminit.c

```diff
@@ -6308,10 +6308,25 @@ parseXlogForCompatibility(J9JavaVM *vm)
 							if ('\0' == *upToComma) {
 								goto xlogerr;
 							}
-							gclog = 0 != j9_cmdla_stricmp("off", upToComma);
-							if (gclog) {
+							/* only levels trace, debug, info, and not off, warning, error, enable the gc verbose log */
+							if (0 == j9_cmdla_stricmp("off", upToComma)) {
+								gclog = FALSE;
+							} else if ((0 == j9_cmdla_stricmp("warning", upToComma))
+								|| (0 == j9_cmdla_stricmp("error", upToComma))
+							) {
+								gclog = FALSE;
 								/* OpenJ9 doesn't understand log levels */
 								unrecognizedOption = TRUE;
+							} else if ((0 == j9_cmdla_stricmp("trace", upToComma))
+								|| (0 == j9_cmdla_stricmp("debug", upToComma))
+								|| (0 == j9_cmdla_stricmp("info", upToComma))
+							) {
+								gclog = TRUE;
+								/* OpenJ9 doesn't understand log levels */
+								unrecognizedOption = TRUE;
```

Comment:
Why do we want this (here and on line 6314)? Isn't part of the point of this change to recognize log levels, and so the warning this implies is no longer appropriate?

Reply:
-Xlog is a Hotspot option, it should only be used on OpenJ9 because of switching from Hotspot to OpenJ9 without updating options. However OpenJ9 doesn't really support -Xlog properly, just a little bit for compatibility. Turning on the gc verbose log isn't really the same as using the options under Hotspot, so I think the warning is appropriate since apps using the options may be expected Hotspot behavior.

Reply:
Ok, that makes sense.